module/Company/src/Company/Mapper/Company.php

```diff
@@ -83,7 +85,12 @@ public function insert($languages)
         $company->setHidden(false);
         $this->em->persist($company);
 
-        return $company;
+        $newcompany = new NewCompanyModel($company);
+        $this->em->persist($newcompany);
+
+        $companies = [$company, $newcompany];
+
+        return $companies;
```

Comment:
Condense to `return [$company, $newcompany];`. Also I would suggest `$newCompany` instead of `$newcompany`.